pkg/database/membership.go

```diff
@@ -31,9 +33,59 @@ type Membership struct {
 	RealmID uint
 	Realm   *Realm
 
+	DefaultSMSTemplateLabel string `gorm:"type:varchar(255);"`
+
 	Permissions rbac.Permission
 }
 
+// SaveMembership saves the membership details. Should have a userID and a realmID to identify it.
+func (db *Database) SaveMembership(m *Membership, actor Auditable) error {
+	if m == nil {
+		return fmt.Errorf("provided membership is nil")
+	}
+
+	if actor == nil {
+		return fmt.Errorf("auditing actor is nil")
+	}
+
+	return db.db.Transaction(func(tx *gorm.DB) error {
+		var audits []*AuditEntry
+
+		var existing Membership
+		if err := tx.
```

Comment:
should we be reading this for update?

Reply:
We do that to diff existing and add audits. This is the pattern we're using in SaveRealm, SaveUser, etc